ocaml/typing/layouts.ml

```diff
@@ -993,16 +996,21 @@ module Layout = struct
 
   let equate = equate_or_equal ~allow_mutation:true
 
+  let score_reason reason =
```

Comment:
Style nitpick: I think it's nicer to use `= function` instead of naming `reason` and then matching on it.